serialization/bigint/marshal_utils.go

```diff
@@ -172,8 +178,14 @@ func EncReflect(v reflect.Value) ([]byte, error) {
 	switch v.Kind() {
 	case reflect.Int, reflect.Int64, reflect.Int32, reflect.Int16, reflect.Int8:
 		return EncInt64(v.Int())
-	case reflect.Uint, reflect.Uint64, reflect.Uint32, reflect.Uint16, reflect.Uint8:
-		return EncUint64(v.Uint())
+	case reflect.Uint32, reflect.Uint16, reflect.Uint8:
+		return encUint64(v.Uint()), nil
+	case reflect.Uint, reflect.Uint64:
+		val := v.Uint()
+		if val > math.MaxInt64 {
+			return nil, fmt.Errorf("failed to marshal bigint: value (%T)(%[1]v) out of range", v.Interface())
+		}
+		return encUint64(val), nil
```

Comment:
why not just call `EncUint64()` here? there is the MaxInt64 check there so no need to duplicate

Reply:
To track exactly where the error occurred, some differences are needed in the errors. 
In this case, these differences - the type of value `(%T)(%[1]v)`.
If you do not do this, it will be more difficult to track the location of the error.

For me, if the content of all errors included the name of the function and something like a `stack trace` , this would be an ideal option.